pkg/operator2/idp.go

```diff
@@ -168,7 +168,7 @@ func (c *authOperator) convertProviderConfigToIDPData(providerConfig *configv1.I
 			return nil, fmt.Errorf(missingProviderFmt, providerConfig.Type)
 		}
 
-		urls, err := c.discoverOpenIDURLs(openIDConfig.Issuer, corev1.ServiceAccountRootCAKey, openIDConfig.CA)
+		urls, challengesAllowed, err := c.discoverOpenIDConfig(openIDConfig.Issuer, corev1.ServiceAccountRootCAKey, openIDConfig.CA)
```

Comment:
nit: Maybe name the boolean more specifically (e.g. 'passwordChallengeSupported') so that it's clear why challenges would be allowed?